opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/measurement_consumer.py

```diff
@@ -90,12 +92,38 @@ def register_asynchronous_instrument(
     def collect(
         self,
         metric_reader: "opentelemetry.sdk.metrics.MetricReader",
+        timeout_millis: float = 10_000,
     ) -> Iterable[Metric]:
+
+        errors = []
+
         with self._lock:
             metric_reader_storage = self._reader_storages[metric_reader]
             # for now, just use the defaults
             callback_options = CallbackOptions()
```

Comment:
Sorry I didn't really handle this before but now seems like a good time–

If the remaining deadline for async callback is less than the default timeout of [10s](https://github.com/open-telemetry/opentelemetry-python/blob/22bfe8223594d459424fa70146a04c11162d5c0f/opentelemetry-api/src/opentelemetry/metrics/_internal/instrument.py#L54), you can pass the remaining deadline to the callback `async_instrument.callback(CallbackOptions(timeout_millis=remaning_deadline_millis))`

Reply:
Fixed